### DISCUSSION TITLE
whl file not importing

### DISCUSSION THREAD

**Discussion by chinmaym505** [pyscript/pyscript#1290] [Q&A] [answered]
when I tried importing a whl file with `<py-config>`, it gave me an error:
`(PY1001): Unable to install package(s) 'webbot-0.34-py3-none-any.whl'. Reason: Traceback (most recent call last): File "/lib/python3.10/site-packages/micropip/_micropip.py", line 161, in _fetch_bytes return await fetch_bytes(self.url, fetch_kwargs) File "/lib/python3.10/site-packages/micropip/_compat_in_pyodide.py", line 30, in fetch_bytes result_bytes = await (await pyfetch(url, **kwargs)).bytes() File "/lib/python3.10/pyodide/http.py", line 254, in pyfetch raise OSError(e.js_error.message) from None OSError: Failed to fetch The above exception was the direct cause of the following exception: Traceback (most recent call last): File "/lib/python3.10/asyncio/futures.py", line 201, in result raise self._exception File "/lib/python3.10/asyncio/tasks.py", line 234, in __step result = coro.throw(exc) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 573, in install await transaction.gather_requirements(requirements) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 333, in gather_requirements await gather(*requirement_promises) File "/lib/python3.10/asyncio/futures.py", line 284, in __await__ yield self # This tells Task to wait for completion. File "/lib/python3.10/asyncio/tasks.py", line 304, in __wakeup future.result() File "/lib/python3.10/asyncio/futures.py", line 201, in result raise self._exception File "/lib/python3.10/asyncio/tasks.py", line 234, in __step result = coro.throw(exc) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 346, in add_requirement await self.add_wheel(wheel, extras=set()) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 461, in add_wheel await wheel.download(self.fetch_kwargs) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 177, in download data = await self._fetch_bytes(fetch_kwargs) File "/lib/python3.10/site-packages/micropip/_micropip.py", line 169, in _fetch_bytes raise ValueError( ValueError: Can't fetch wheel from 'webbot-0.34-py3-none-any.whl'. One common reason for this is when the server blocks Cross-Origin Resource Sharing (CORS). Check if the server is sending the correct 'Access-Control-Allow-Origin' header. . Please open an issue at https://github.com/pyscript/pyscript/issues/new if you require help or you think it's a bug.`
Is there any way to fix this?

The whl file is with the HTML document

**Comment by JeffersGlass** [comment]
Can you share the contents of your `<py-config>` tag?

Additionally, what version of PyScript are you using? I.e., what is the link your `<script>` tag points to?

**Comment by yellowbean** [comment]
It would be great to have a example code for this ? 

it doesn't work for me either.

here is the folder structure

![Screenshot 2024-03-21 at 22 12 34](https://github.com/pyscript/pyscript/assets/1008321/241a8231-b28d-4857-9797-0018ff625d98)

here is the index.html

    <head>
        <title>Jolly Frog</title>
        
        <script type="module" src="https://pyscript.net/releases/2024.3.1/core.js"></script>
    </head>
    <body>
        <py-env>
            - ./dist/abc.whl
        </py-env>

thanks to @mudream4869 which gave an example
https://github.com/mudream4869/pyscript-local-package

But seems the syntax is deprecated ?  it would be great to have an example for emfs ? 

![Screenshot 2024-03-21 at 22 15 36](https://github.com/pyscript/pyscript/assets/1008321/366860be-15f1-44a0-b867-6b334e8a7328)

**Reply by WebReflection** [reply]
`<py-env>` doesn't exist since about a year or so ... and I am sure @JeffersGlass can help you further but **no py-env** in current PyScript.

**Reply by WebReflection** [reply]
now that I read it further, the `emfs://` protocol is AFAIK not supported right now but it's also the first time I hear about it ... I don't know what it is or what it does but then again, your `.whl` should be listed in `packages` entry within a `<py-config>` TOML or JSON tag or as separate external file to point at via `config="my-config.toml"` attribute within the `<script type="py" config="...">` container for your Python code.

That repository is 2 years old and I'd like to see it either archived or changed to provide instructions to the latest PyScript, although that should indeed be more likely part of our documentation /cc @ntoll @mudream4869

**Reply by mudream4869** [reply]
Thanks for reminding me 😄
I have updated the repo to works with 2024.3.1 ver pyscript~

**Comment by yellowbean** [comment, marked as answer]
the solution so far works for me:

in pyscript.toml

    packages=["arrr"
             ,"http://localhost:8001/dist/abc-0.1.1-py3-none-any.whl"]

**Reply by WebReflection** [reply]
as the repository you initially took the example from is working with a relative path, I elieve you can go back to just `./dist/abc-........whl` instead and still see it working: https://github.com/mudream4869/pyscript-local-package?tab=readme-ov-file#files

I will close this as it's resolved.

**Comment by WebReflection** [comment]
Closing as working as expected and updated the remote example https://github.com/mudream4869/pyscript-local-package?tab=readme-ov-file#files - thanks @mudream4869
